File: problem15.php
<?php
//solution: 137846528820
echo eulerProblem15() . "\n";

function eulerProblem15() {
    return combinations(40, 20);
}

function combinations($n, $k) {
    return factorial($n) / (factorial($k) * factorial($n - $k));
}

function factorial($num) {
    $fact = 1;
    for($i = 2; $i <= $num; ++$i) {
        $fact *= $i;
    }
    return $fact;
}

?>

File: python/problem15.py
#!/usr/bin/python
#solution: 137846528820

def factorial(n):
    if n == 2:
        return 2
    return n * factorial(n - 1)

def combinations(n, r):
    return factorial(n) / (factorial(r) * factorial(n - r))

def euler_problem_15():
    return combinations(40, 20)

print euler_problem_15()
